src/createElement.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * 일반 태그이거나 함수형 컴포넌트면 createElement 함수 실행
+ */
+export const createElement = (type: string | Function, props: JSX.Props, ...children: any[]) => {
+  // 함수형 컴포넌트면 props랑 전개한 children 내려줘서 재귀적으로 createElement 실행
+  if (typeof type === 'function') {
+    return type(props, ...children);
+  }
+
+  return {
+    type,
+    props: {
+      ...props,
+      children:
+        children.length === 1 // children이 하나만 있으면 배열 말고 객체로 처리
+          ? validTextNode(children[0])
+          : children.map((child) => validTextNode(child)), // children이 여러개면 배열로 묶어서 처리
+    },
+  };
+};
+
+/**
+ * leaf 노드에 있는 텍스트 노드면 type: createTextNode로 처리
+ */
+const validTextNode = (child: any) => {
+  if (typeof child === 'string' || typeof child === 'number' || typeof child === 'boolean') {
+    return createTextNode(child);
+  }
+  return child;
+};
+
+/**
+ * leaf 노드에 있는 값이 string, number, boolean이면 textNode로 표시
+ */
+const createTextNode = (child: string | number | boolean) => {
+  return { type: 'textNode', children: child };
+};
+
+/**
+ * fragment 태그면 Fragment 함수 자동으로 실행
+ */
+export const Fragment = (props: any, ...children: any[]) => {
+  return { type: 'fragment', props: { ...props, children: [...children] } };
+};
```

Comment:
```suggestion
export const Fragment = (props: any, ...children: any[]) => {
  return { type: 'fragment', props: { ...props, children } };
};
```
매개변수를 받을 때 `...children` rest 문법으로 받아서 이미 얕은 복사가 되었을거라고 생각이 되는데,
한번 더 `children: [...children]` 으로 얕은 복사를 수행하신 이유가 있으신가요??

Reply:
해당 부분 좀 다른 방식으로 처리하게 코드 수정해서 다시 커밋 올려뒀습니다! 생각해보니 Fragment일 때에도 자식 요소가 HTMLElement인지 textNode인지 확인하는 절차가 필요해서, createElement의 return 부분이랑 type 프로퍼티 값만 다르지 사실 같은 로직으로 처리가 돼야했네요

Reply:
오호 그러네요 저도 열심히 리팩토링 해봐야겠네요ㅋㅋㅋ